provider/resources.go

```diff
@@ -4854,7 +4838,7 @@ func Provider() tfbridge.ProviderInfo {
 	prov.SetAutonaming(255, "-")
 
 	// Add a CSharp-specific override for aws_s3_bucket.bucket.
-	prov.Resources["aws_s3_bucket"].Fields["bucket"].CSharpName = "BucketName"
+	prov.Resources["aws_s3_bucket_legacy"].Fields["bucket"].CSharpName = "BucketName"
```

Comment:
Comment above should explain why we are doing this... and now that I think of it, why aren't we using `CSharpName` inline instead?